master/internal/resourcemanagers/resource_pool.go

```diff
@@ -322,15 +323,36 @@ func (rp *ResourcePool) releaseResource(ctx *actor.Context, handler *actor.Ref)
 	handler.System().Tell(handler, sproto.ReleaseResources{ResourcePool: rp.config.PoolName})
 }
 
-func (rp *ResourcePool) resourcesReleased(ctx *actor.Context, handler *actor.Ref) {
-	if allocated := rp.taskList.GetAllocations(handler); allocated != nil {
-		ctx.Log().Infof("resources are released for %s", handler.Address())
-		for _, allocation := range allocated.Resources {
-			typed := allocation.(*containerResources)
-			ctx.Tell(typed.agent.Handler, agent.DeallocateContainer{ContainerID: typed.containerID})
+func (rp *ResourcePool) resourcesReleased(
+	ctx *actor.Context,
+	msg sproto.ResourcesReleased,
+) {
+	a := rp.taskList.GetAllocations(msg.TaskActor)
+	if a == nil {
+		return
+	}
+
+	switch {
+	case msg.ResourcesID != nil:
+		ctx.Log().Infof("resource %v is released for %s", msg.ResourcesID, msg.TaskActor.Address())
+		for i, r := range a.Resources {
+			if r.Summary().ResourcesID == *msg.ResourcesID {
+				continue
+			}
+
+			cr := r.(*containerResources)
+			ctx.Tell(cr.agent.Handler, agent.DeallocateContainer{ContainerID: cr.containerID})
+			slices.Delete(a.Resources, i, i+1)
```

Comment:
this is mildly scary but can't say anything, probably works

Reply:
best way to delete from a slice in go as far as I know. just does the append(xs[:I], xs[i+1]...) slice trick internally